File: LeetCode/MockInterviews/C#/PalindromeNumber.cs
using System;

namespace MockInterviews
{
    public class PalindromeNumber
    {

        public static bool IsPalindrome(int x) {
            int number = 0;
            int res = 0;
            int limit = x;
            while (limit > 0) {
                res = limit % 10;
                limit -= res;
                limit /= 10;
                number *= 10;
                number += res;
            }
            return number == x;
        }

        public static void Main(string[] args)
        {
            Console.WriteLine(IsPalindrome(123321));
            Console.WriteLine(IsPalindrome(1234321));
            Console.WriteLine(IsPalindrome(123456));
            Console.WriteLine(IsPalindrome(123421));
        }
    }
}

File: LeetCode/MockInterviews/Python/palindrome_number.py

def is_palindrome(number):
    flip = number
    res = 0
    pal = 0
    while flip > 0:
        res = flip % 10
        flip -= res
        flip /= 10
        pal *= 10
        pal += res
    return pal == number

if __name__ == '__main__':
    print(is_palindrome(123321))
    print(is_palindrome(12321))
    print(is_palindrome(12345))
